sites/all/modules/unl_multisite/cron.php

```diff
@@ -297,19 +298,109 @@ function unl_add_site($site_path, $uri, $clean_url, $db_prefix, $site_id) {
     throw new Exception('Error while running drush site-install.');
   }
 
+  if ($clone_from_id) {
+    unl_clone_site($clone_from_id, $site_id);
+  }
+
   unl_add_site_to_htaccess($site_id, $site_path, FALSE);
 }
 
-function unl_remove_site($site_path, $uri, $db_prefix, $site_id) {
-  // Grab the list of tables we need to drop.
-  $schema = drupal_get_schema(NULL, TRUE);
-  $tables = array_keys($schema);
-  sort($tables);
+function unl_get_site_record($site_id) {
+  $query = db_query('SELECT * FROM {unl_sites} WHERE site_id='.(int)$site_id);
 
+  return $query->fetchAssoc();
+}
+
+/**
+ * @param $from_id
+ * @param $to_id
+ *
+ * @throws Exception
+ */
+function unl_clone_site($from_id, $to_id) {
+  //Get the sites to clone
+  if (!$clone_from_site = unl_get_site_record($from_id)) {
+    //Unable to get the site to clone from
+    throw new Exception('Error while cloning site. Site id ' . $from_id . ' could not be found');
+  }
+
+  if (!$clone_to_site = unl_get_site_record($to_id)) {
+    //Unable to get the site to clone to
+    throw new Exception('Error while cloning site. Site id ' . $to_id . ' could not be found');
+  }
+  
+  //Delete the to site
+  unl_drop_site_tables($clone_to_site['db_prefix']);
+  
+  //Copy tables
   $database = $GLOBALS['databases']['default']['default'];
-  $db_prefix .= '_' . $database['prefix'];
+  $from_db_prefix = $clone_from_site['db_prefix'] .'_' . $database['prefix'];
+  $to_db_prefix = $clone_to_site['db_prefix'] .'_' . $database['prefix'];
+  
+  // Grab the list of tables we need to clone.
+  $tables = getTableNamesWithPrefix($from_db_prefix);
+
+  // Clone the site's tables
+  foreach ($tables as $table) {
+    $from_table = $table;
+    $to_table = str_replace_once($from_db_prefix, $to_db_prefix, $table);
```

Comment:
it is a function, just not one that you have. :P This should be fixed now, and perhaps I shouldn't blindly auto-complete function names in the future.